addons/binding/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVHandlerFactory.java

```diff
@@ -0,0 +1,347 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.magentatv.internal;
+
+import static org.openhab.binding.magentatv.internal.MagentaTVBindingConstants.*;
+
+import java.net.NetworkInterface;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.net.NetworkAddressService;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.openhab.binding.magentatv.internal.network.MagentaTVNetwork;
+import org.openhab.binding.magentatv.internal.network.MagentaTVPoweroffListener;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link MagentaTVHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author markus7017 - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = { ThingHandlerFactory.class,
+        MagentaTVHandlerFactory.class }, immediate = true, configurationPid = "binding." + BINDING_ID)
+public class MagentaTVHandlerFactory extends BaseThingHandlerFactory {
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_RECEIVER);
+    private final MagentaTVLogger logger = new MagentaTVLogger(MagentaTVHandlerFactory.class, "Factory");
+    private @Nullable NetworkAddressService networkAddressService = null;
+
+    private final MagentaTVNetwork network = new MagentaTVNetwork();
+    private @Nullable MagentaTVPoweroffListener upnpListener;
+    private boolean notifyServletInitialized = false;
+
+    protected class MagentaTVDevice {
```

Comment:
Use static classes. Otherwise the class will have a reference to the factory which might cause memory leaks in the end.

Reply:
need to check